apps/server/src/modules/board/controller/dto/submission-item/submission-item.response.ts

```diff
@@ -0,0 +1,23 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { TimestampsResponse } from '../timestamps.response';
+
+export class SubmissionItemResponse {
+	constructor({ id, timestamps, completed, userId }: SubmissionItemResponse) {
+		this.id = id;
+		this.timestamps = timestamps;
+		this.completed = completed;
+		this.userId = userId;
+	}
+
+	@ApiProperty({ pattern: '[a-f0-9]{24}' })
+	id: string;
+
+	@ApiProperty()
+	timestamps: TimestampsResponse;
+
+	@ApiProperty()
+	completed: boolean;
```

Comment:
A SubmissionItem sounds very general. How will responses and data-model look, when there are other types of submissions ... like SubissionWith written text... or submission with uploaded files? Will you simply add those properties and allow all of them to be answered... depending what the teacher allows/demands?

Reply:
that is the general (but for now vague) idea, to expand it so that it can handle additional types of content. But how exactly, is not part of the PR.

Submission will be more strict compared to elements. For example, you cannot add unlimited files or rich-text. We will have e.g. a submission grade (which only the teacher can fill). Possibly, one file-upload, one normal rich-text, another rich-text for comments, etc. All of that, should to be part of a submission-item. 
If they can be props, traits, or something else, we'll be clarified elsewhere.